spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/expressions/TransformExpressions.scala

```diff
@@ -119,3 +118,38 @@ case class IcebergBucketTransform(numBuckets: Int, child: Expression) extends Ic
 
   override def dataType: DataType = IntegerType
 }
+
+case class IcebergTruncateTransform(child: Expression, width: Int) extends IcebergTransformExpression {
+
+  override def children: Seq[Expression] = child :: Nil
+
+  @transient lazy val truncateFunc: Any => Any = child.dataType match {
+    case _: DecimalType =>
+      val t = Transforms.truncate[Any](icebergInputType, width)
+      d: Any =>  Decimal.apply(t(d.asInstanceOf[Decimal].toJavaBigDecimal).asInstanceOf[java.math.BigDecimal])
+    case _: StringType =>
+      val t = Transforms.truncate[String](Types.StringType.get(), width)
+      s: Any => UTF8String.fromString(t(s.toString))
+    case _: BinaryType =>
+      val t = Transforms.truncate[ByteBuffer](Types.BinaryType.get(), width)
+      s: Any => {
+        val res: ByteBuffer = t(ByteBuffer.wrap(s.asInstanceOf[Array[Byte]]))
+        val returnValue: Array[Byte] = new Array[Byte](res.limit())
+        res.rewind();
+        res.get(returnValue);
```

Comment:
@rdblue / @aokolnychyi Is there a better way to do this here ?

Reply:
Yes, I recommend using `ByteBuffers.toByteArray`. That will handle the conversion back to byte array efficiently.